File: pset6/credit/credit.py
from cs50 import get_int

# Get number as int from user
intNumber = get_int("Long Card Number ")

# Convert int to list
number = list(map(int, str(intNumber)))

# Get length of list
numberlen = len(number)

# Check numberlen is within range
if numberlen < 13 or numberlen > 16:
    print("INVALID")
    exit()

# Sum the numbers
intsum = 0

if numberlen == 13 or numberlen == 15:
    a = 1
    b = 0
else:
    a = 0
    b = 1

for i in range(a, numberlen, 2):

    numberi = number[i] * 2

    if numberi < 10:
        intsum = intsum + numberi
    elif numberi >= 10:
        intsum = intsum + 1 + (numberi - 10)
    else:
        print("ERROR")
        exit()

for i in range(b, numberlen, 2):
    intsum = intsum + number[i]

# Convert first two digits of number to an integer
startdigits = (number[0] * 10) + (number[1])

# Check final sum number, compare numberlen to expected length of card, return type
if intsum % 10 == 0:
    if startdigits == 34 or startdigits == 37:
        print("AMEX")
        exit()
    elif startdigits > 50 and startdigits < 56 and numberlen == 16:
        print("MASTERCARD")
        exit()
    elif startdigits > 39 and startdigits < 50 and (numberlen == 13 or numberlen == 16):
        print("VISA")
        exit()
    else:
        print("INVALID")
        exit()

else:
    print("INVALID")
    exit()

File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main(void)
{
    //Get Number
    long intnumber = get_long("Long Card Number: ");

    //Convert intnumber to a string
    char number[20];
    sprintf(number, "%li", intnumber);

    //Get length of string
    int numberlen = strlen(number);

    //Check numberlen is within range
    if (numberlen < 13 || numberlen > 16)
    {
        printf("INVALID\n");
        return 0;
    }

    //Sum the numbers
    int intsum = 0;

    int a = 0;
    int b = 1;

    if (numberlen == 13 || numberlen == 15)
    {
        a = 1;
        b = 0;
    }

    for (int i = a; i < numberlen; i += 2)
    {

        int numberi = number[i] - 48;
        numberi = numberi * 2;

        if (numberi < 10)
        {
            intsum = intsum + (numberi);
        }

        else if (numberi >= 10)
        {
            intsum = intsum + 1 + (numberi - 10);
        }

        else
        {
            printf("Error");
            return 1;
        }

    }

    for (int i = b; i < numberlen; i += 2)
    {
        intsum = intsum + (number[i] - 48);
    }

    //Convert First two digits of number to an integer
    int startdigits = ((number[0] - 48) * 10) + (number[1] - 48);

    //Check final sum number is 0,
    //compare numberlen to expected length of relevant card-type,
    //if valid, return type, else return invalid
    if (intsum % 10 == 0)
    {
        if (startdigits == 34 || startdigits == 37)
        {
            printf("AMEX\n");
            return 0;
        }

        else if (startdigits > 50 && startdigits < 56 && numberlen == 16)
        {
            printf("MASTERCARD\n");
            return 0;
        }

        else if (startdigits > 39 && startdigits < 50 && (numberlen == 13 || numberlen == 16))
        {
            printf("VISA\n");
            return 0;
        }

        else
        {
            printf("INVALID\n");
            return 0;
        }

    }

    else
    {
        printf("INVALID\n");
        return 0;
    }

}